Reading wannier90 system with guiding_centers = true can fail

If `guiding_centers = true` is used in Wannier90, reading system with `transl_inv=True` can fail because the assert is False https://github.com/wannier-berri/wannier-berri/blob/9f2515fc0f55b33baecc88461f465a5b7de6ad05/wannierberri/__system_w90.py#L97-L101

The reason is that `chk.get_AA_q` does not use the guiding center method to calculate the Wannier centers, while the Wannier90 output (`wannier_centers_cart_auto`) is calculated using guiding centers.

(In most cases, guiding_centers does nothing, so the current implementation works. But there are some edge cases, like monolayers placed at the center of the supercell, that guiding_centers have significant effects. I attached an example.)

A simple workaround is just to trust the Wannier centers calculated by Wannier90, and overwrite the diagonals.
```python
if transl_inv:
    for iw in range(self.num_wann):
        self.AA_R[iw, iw, self.iR0, :] = self.wannier_centers_cart_auto[iw, :]
```

A complete fix will be to implement guiding centers in `__w90_files.py`, but I do not understand the theory of guiding centers so I cannot say whether it will be easy or not.

Reproducer (unzip and run `./run.sh`):
[input.zip](https://github.com/wannier-berri/wannier-berri/files/7553059/input.zip)

Wannier-Berri output:
```
[jmlim@im0 temp]$ ./wberri.py 
using fortio to read
Reading restart information from file BN.chk :
Time to read .chk : 0.015824317932128906
using ws_distance
irvec_new_all shape (43,)
time for ws_dist_map :  0.05253195762634277 0.023316383361816406 0.020260095596313477 0.007009744644165039 0.0019457340240478516
Time for MMN.__init__() : 0.0742490291595459 , read : 0.06488180160522461 , headstring 0.009367227554321289
Traceback (most recent call last):
  File "./wberri.py", line 3, in <module>
    system = wberri.System_w90('BN', berry=True, transl_inv=True)
  File "/group2/jmlim/shift_phase_convention/wannier-berri/wannierberri/__system_w90.py", line 101, in __init__
    wannier_centers_cart_new,self.wannier_centers_cart_auto-wannier_centers_cart_new))
AssertionError: the difference between read
[[ 1.2515 -0.7227  7.5003]
 [ 1.2515 -0.7226  7.5003]
 [ 1.2515 -0.8385  7.5003]
 [ 1.2515 -0.6032  7.5003]]
 and evluated 
[[ 1.2515e+00+0.j -7.2268e-01+0.j  4.7395e-10+0.j]
 [ 1.2515e+00+0.j -7.2256e-01+0.j -1.2500e+00+0.j]
 [ 1.2515e+00+0.j -8.3848e-01+0.j  8.3336e-01+0.j]
 [ 1.2515e+00+0.j -6.0317e-01+0.j -1.2500e+00+0.j]]
 wannier centers is 
[[-5.0069e-12+0.j  2.8912e-12+0.j  7.5003e+00+0.j]
 [-5.0069e-12+0.j  2.8907e-12+0.j  8.7503e+00+0.j]
 [-5.0069e-12+0.j  3.3399e-12+0.j  6.6669e+00+0.j]
 [-5.0067e-12+0.j  2.4308e-12+0.j  8.7503e+00+0.j]]
```

--- a/wannierberri/__system_w90.py
+++ b/wannierberri/__system_w90.py
@@ -99,7 +99,6 @@ class System_w90(System):
             if transl_inv:
                 wannier_centers_cart_new = np.diagonal(self.AA_R[:,:,self.iR0,:],axis1=0,axis2=1).transpose()
                 if not np.all(abs(wannier_centers_cart_new-self.wannier_centers_cart_auto)<1e-6):
-                    print(f"{guiding_centers=}")
                     if guiding_centers:
                         print(f"The read Wannier centers\n{self.wannier_centers_cart_auto}\n"
                               f"are different from the evaluated Wannier centers\n{wannier_centers_cart_new}\n"

--- a/wannierberri/__system_w90.py
+++ b/wannierberri/__system_w90.py
@@ -40,6 +40,9 @@ class System_w90(System):
         the seedname used in Wannier90
     transl_inv : bool
         Use Eq.(31) of `Marzari&Vanderbilt PRB 56, 12847 (1997) <https://journals.aps.org/prb/abstract/10.1103/PhysRevB.56.12847>`_ for band-diagonal position matrix elements
+    guiding_centers : bool
+        If True, enable overwriting the diagonal elements of the AA_R matrix at R=0 with the
+        Wannier centers calculated from Wannier90.
     npar : int
         number of processes used in the constructor
     fft : str
@@ -53,6 +56,7 @@ class System_w90(System):
 
     def __init__(self,seedname="wannier90",
                     transl_inv=True,
+                    guiding_centers=False,
                     fft='fftw',
                     npar=multiprocessing.cpu_count()  , 
                     **parameters
@@ -94,9 +98,21 @@ class System_w90(System):
             timeFFT+=time()-t0
             if transl_inv:
                 wannier_centers_cart_new = np.diagonal(self.AA_R[:,:,self.iR0,:],axis1=0,axis2=1).transpose()
-                assert np.all(abs(wannier_centers_cart_new-self.wannier_centers_cart_auto)<1e-6), (
-                  "the difference between read\n{}\n and evluated \n{}\n wannier centers is \n{}\n".format(self.wannier_centers_cart_auto,
-                        wannier_centers_cart_new,self.wannier_centers_cart_auto-wannier_centers_cart_new))
+                if not np.all(abs(wannier_centers_cart_new-self.wannier_centers_cart_auto)<1e-6):
+                    print(f"{guiding_centers=}")
+                    if guiding_centers:
+                        print(f"The read Wannier centers\n{self.wannier_centers_cart_auto}\n"
+                              f"are different from the evaluated Wannier centers\n{wannier_centers_cart_new}\n"
+                              "This can happen if guiding_centres was set to true in Wannier90.\n"
+                              "Overwrite the evaluated centers using the read centers.")
+                        for iw in range(self.num_wann):
+                            self.AA_R[iw, iw, self.iR0, :] = self.wannier_centers_cart_auto[iw, :]
+                    else:
+                        raise ValueError(
+                            f"the difference between read\n{self.wannier_centers_cart_auto}\n"
+                            f"and evluated \n{wannier_centers_cart_new}\n wannier centers is\n"
+                            f"{self.wannier_centers_cart_auto-wannier_centers_cart_new}\n"
+                            "If guiding_centres was set to true in Wannier90, pass guiding_centers = True to System_w90.")
 
 
         if self.getBB:

